saltproc/input_schema.json

```diff
@@ -0,0 +1,121 @@
+                    "type": "string" },
+                "npop": {
+                    "description": "Number of neutrons per generation",
+                    "type": "number",
+                    "minimum": 0},
+                "active_cycles": {
+                    "description": "number of active generations",
+                    "type": "number",
+                    "minimum": 0},
+                "inactive_cycles": {
+                    "description": "Number of inactive generations",
+                    "type": "number",
+                    "minimum": 0},
+                "geo_file_paths": {
+                    "description": "Path(s) to geometry file(s) to swtich to in depletion code runs",
+                    "type": "array",
+                    "items": { "type": "string"},
+                    "minItems": 1,
+                    "uniqueItems": false
+                }
+            },
+            "required": ["codename", "exec_path", "template_inputfile_path", "iter_inputfile", "iter_matfile", "npop", "active_cycles","inactive_cycles", "geo_file_paths"]
+        },
+       "simulation": {
+           "description": "Simulation class input parameters",
+           "type": "object",
+           "properties": {
+               "sim_name": {
+                   "description": "Name of simulation",
+                   "type": "string"},
+               "db_name": {
+                   "description": "Output HDF5 database file name",
+                   "type": "string",
+                    "pattern": "^(.*)\\.h5$"},
+               "restart_flag": {
+                   "description": "Restart simulation from the step when it stopped?",
+                   "type": "boolean"},
+               "adjust_geo": {
+                   "description": "switch to another geometry when keff drops below 1?",
+                   "type": "boolean"}
+           },
+           "requires": ["sim_name", "db_name", "restart_flag", "adjust_geo"]
+       },
+       "reactor": {
+           "description": "Reactor class input parameters",
+           "type": "object",
+           "properties": {
+               "volume": {
+                   "description": "reactor core volume [cm^3]",
+                   "type": "number",
+                    "minimum": 0},
+               "mass_flowrate": {
+                   "description": "Salt mass flowrate through reactor core [g/s]",
+                   "type": "number",
+                    "minimum": 0 },
+               "power_levels": {
+                   "description": "Reactor power or power step list durng depletion step [W]",
+                   "type": "array",
+                   "items": { "type": "number",
+                    "minimum": 0},
+                   "minItems": 1,
+                   "uniqueItems": false
+                },
+               "depl_hist": {
+                   "description": "Depletion step interval or cumulative time (end of step) [d]",
```

Comment:
I'm gonna change `depsteps` to `num_depsteps` and `depl_hist` to `dep_step_len`. I'll need to change the code a bit but better to do this now since we're changing the API anyways.